test/e2e/components/vl-map.js

```diff
@@ -1,6 +1,70 @@
 const { VlElement } = require('vl-ui-core').Test;
+const { By } = require('vl-ui-core').Test.Setup;
+const VlMapBaseLayer = require('./vl-map-baselayer');
+const VlMapSearch = require('./vl-map-search');
 
-class VlMap extends VlElement {  
+class VlMap extends VlElement {
+	async _getMap() {
+		return this.shadowRoot;
+	}
+
+	async getBaseLayers() {
+		const childElements = await this.findElements(By.css(this.selector + ' *'));
+		const tagNames = await Promise.all(childElements.map(element => element.getTagName()));
+		const baseLayerElements = childElements.filter((element, index) => tagNames[index].indexOf('vl-map-baselayer') > -1);
+		return Promise.all(baseLayerElements.map(async element => await new VlMapBaseLayer(this.driver, element)));
+	}
+	
+	async isEscapeKeyDisabled() {
+		return this.hasAttribute("disable-escape-key");
+	}
+
+	async isRotationDisabled() {
+		return this.hasAttribute("disable-rotation");
+	}
+	
+	async isMouseWheelZoomDisabled() {
+		return this.hasAttribute("disable-mouse-wheel-zoom");
+	}
+
+	async getOverviewMap() {
+		const map = await this._getMap();
+		await this.driver.wait(async () => {
+            const overviewMaps = await map.findElements(By.css('.ol-overviewmap'));
+			return overviewMaps.length > 0;
+		});
+		return await map.findElement(By.css('.ol-overviewmap'));
+	}
+
+	async getActiveBaseLayerTitle() {
+        return this.driver.executeScript(`
+			return arguments[0].map.baseLayers.find((layer) => {
+				return layer.getVisible();
+			}).get('title')
+				`, this);
+    }
+
+	async toggleBaseLayer() {
+		const overviewMap = await this.getOverviewMap();
+		return overviewMap.click();
+	}
+
+    async hasSearch() {
+        const search = await this.driver.executeScript(`return arguments[0].shadowRoot.querySelector('vl-map-search')`, this);
```

Comment:
Lijkt me iets wat ook vanuit selenium kan?

Reply:
Met die shadowRoot weet ik het niet zo goed, dan zou daar eerst een element van gemaakt moeten worden en vermoed niet dat in firefox werkt. Dat is toch waarom lastElementChild in VlElement wordt gebruikt dacht ik?

Reply:
Ik heb in verschillende componenten al this.shadowroot.findElement('vl-ui-foobar') gedaan, dus zou moeten werken. Of is er iets speciaals in map waardoor dit niet werkt?

Reply:
Juist nog eens getest, en door die lastElementChild implementatie in VlElement lijkt de vl-map-search niet vindbaar